src/hash_matrix.rs

```diff
@@ -32,15 +33,53 @@ impl ToBigUint for U256 {
 pub struct HashMatrix(u128, u128, u128, u128);
 
 impl HashMatrix {
+    #[must_use]
+    #[inline]
+    pub(crate) fn generic_array_digest(&self) -> generic_array::GenericArray<u8, U64> {
+        digest::generic_array::GenericArray::from_exact_iter(
+            [self.0, self.1, self.2, self.3]
+                .iter()
+                .flat_map(|x| x.to_be_bytes()),
+        )
+        .unwrap()
+    }
+}
+
+trait DigestString {
+    fn to_hex(self) -> String;
+}
+
+impl DigestString for HashMatrix {
     /// Produce a hex digest of the hash. This will be 128 hex digits.
     #[must_use]
     #[inline]
-    pub fn to_hex(self) -> String {
-        format!("{:016x}{:016x}{:016x}{:016x}",
+    fn to_hex(self) -> String {
+        format!("{:032x}{:032x}{:032x}{:032x}",
```

Comment:
I fixed the string digest, it wouldn't always produce 128 hex digit long strings. :sweat_smile: